src/algebra/ring/ulift.lean

```diff
@@ -25,6 +24,14 @@ variables {α : Type u} {x y : ulift.{v} α}
 
 namespace ulift
 
+instance [has_nat_cast α] : has_nat_cast (ulift α) := ⟨λ n, up n⟩
+instance [has_int_cast α] : has_int_cast (ulift α) := ⟨λ n, up n⟩
+
+@[simp, norm_cast] lemma up_nat_cast [has_nat_cast α] (n : ℕ) : up (n : α) = n := rfl
+@[simp, norm_cast] lemma up_int_cast [has_int_cast α] (n : ℤ) : up (n : α) = n := rfl
+@[simp, norm_cast] lemma down_nat_cast [has_nat_cast α] (n : ℕ) : down (n : ulift α) = n := rfl
+@[simp, norm_cast] lemma down_int_cast [has_int_cast α] (n : ℤ) : down (n : ulift α) = n := rfl
```

Comment:
These belong (and are already in?) `algebra.group.ulift`.

Reply:
What? Surely not. `nat.cast` and `int.cast` come from `add_monoid_with_one` and `add_group_with_one` respectively,; which are ring-like.

Reply:
Moved for now

Reply:
I guess what I mean is "these belong there because the instances already are there. I wouldn't be opposed to a move, but I think it would just be churn.